collector/textfile.go

```diff
@@ -308,7 +307,12 @@ fileLoop:
 
 		// Only set this once it has been parsed and validated, so that
 		// a failure does not appear fresh.
-		mtimes[f.Name()] = f.ModTime()
+		info, err := f.Info()
+		if err != nil {
+			error = 1.0
```

Comment:
It's be worth adding a log entry here, to prevent user confusion in the event of a failure.